src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/lib/snippets/insert_trigger_snippet.ts

```diff
@@ -7,47 +7,131 @@
+    triggersKey.range
+      ? getMonacoRangeFromYamlRange(model, triggersKey.range as Range)
+      : null;
+
+  const indentLevel = triggersKeyRange
+    ? getIndentLevelFromLineNumber(model, triggersKeyRange.startLineNumber) + 2
+    : 2;
+
+  return { range: triggersKeyRange, indentLevel };
+}
+
+/**
+ * Creates a replacement range for an empty item line
+ */
+function createReplacementRange(model: monaco.editor.ITextModel, lineNumber: number): monaco.Range {
+  const nextLineNumber = lineNumber + 1;
+  if (nextLineNumber <= model.getLineCount()) {
+    return new monaco.Range(lineNumber, 1, nextLineNumber, 1);
+  }
+  const lineEndColumn = model.getLineMaxColumn(lineNumber);
+  return new monaco.Range(lineNumber, 1, lineNumber, lineEndColumn);
+}
 
+// Algorithm:
+// 1. Check if triggers section exists (even if empty or has empty items)
+// 2. If triggers section exists, find the next line after the last trigger node range
+// 3. If no trigger section found, add triggers: section in the first line of yaml
 export function insertTriggerSnippet(
   model: monaco.editor.ITextModel,
-  yamlDocument: Document,
+  yamlDocument: Document | null,
   triggerType: TriggerType,
   editor?: monaco.editor.IStandaloneCodeEditor
 ) {
-  // find triggers: line number and column number
-  const triggerNodes = getTriggerNodes(yamlDocument);
+  const document = yamlDocument || parseDocument(model.getValue());
+
+  const triggerNodes = getTriggerNodes(document);
   const triggerNode = triggerNodes.find((node) => node.triggerType === triggerType);
-  let insertTriggersSection = false;
-  let insertAtLineNumber = 1;
-  let indentLevel = 0;
   if (triggerNode) {
     // do not override existing trigger
     return;
   }
-  if (triggerNodes.length > 0) {
-    const lastTriggerRange = getMonacoRangeFromYamlNode(
+
+  const triggersPair = getTriggersPair(document);
+  let insertTriggersSection = true;
+  let insertAtLineNumber = 1;
+  let replaceRange: monaco.Range | null = null;
+  let indentLevel = 0;
+
+  if (triggersPair) {
+    insertTriggersSection = false;
+    const { range: triggersKeyRange, indentLevel: expectedIndent } = getTriggersKeyInfo(
       model,
-      triggerNodes[triggerNodes.length - 1].node
+      triggersPair
     );
-    if (lastTriggerRange) {
-      // add a newline after the last trigger
-      insertAtLineNumber = lastTriggerRange.endLineNumber;
-      indentLevel = getIndentLevelFromLineNumber(model, lastTriggerRange.startLineNumber);
+
+    // If triggers section is empty (no triggers with type field), replace the empty content
+    if (
+      triggerNodes.length === 0 &&
+      triggersPair.value &&
+      isSeq(triggersPair.value) &&
+      triggersPair.value.items.length > 0
+    ) {
+      const lastItem = triggersPair.value.items[triggersPair.value.items.length - 1];
+      let emptyItemLineNumber: number | null = null;
+
+      if (lastItem && typeof lastItem === 'object' && 'range' in lastItem && lastItem.range) {
```

Comment:
this condition can be simplified using type guard utils: 
```suggestion
      if (isNode(lastItem) && lastItem.range) {
```

where `isNode` is imported from `yaml`